pym/portage/meta_manifest.py

```diff
@@ -0,0 +1,270 @@
+			for f in files:
+				try:
+					f = _unicode_decode(f,encoding=_encodings['fs'], errors='strict')
+					eclass_dir = _unicode_decode(eclass_dir,encoding=_encodings['fs'], errors='strict')
+				except UnicodeDecodeError:
+									continue
+				fpath = os.path.join(eclass_dir, f)
+				ftype = guessManifestFileType(fpath, is_pkg=False)
+				f = fpath.replace(self.pkgdir, "")
+				if not f.endswith("Manifest"):
+					self.fhashdict[ftype][f] = perform_multiple_checksums(fpath, self.hashes)
+
+	def create_cat(self):
+		'''Creates a MetaManifest file in the selected category'''
+		catdir = self.pkgdir
+		for ftype in MANIFEST2_IDENTIFIERS:
+						self.fhashdict[ftype] = {}
+		for pkg_dir in os.listdir(catdir):
+			pkg_dir = os.path.join(catdir, pkg_dir)
+			if os.path.isdir(pkg_dir):
+				for file in os.listdir(pkg_dir):
+					fpath = os.path.join(pkg_dir, file)
+					if os.path.isfile(fpath) and fpath.endswith('Manifest'):
+						new_mf = MetaManifest(os.path.dirname(fpath))
+						new_mf.verify_manifest()
+						f = fpath.replace(self.pkgdir, "")
+						self.fhashdict["MANIFEST"][f] = perform_multiple_checksums(fpath, self.hashes)
+
+	def create_profile(self):
+		'''Creates a MetaManifest file in the profiles directory'''
+		profiledir = self.pkgdir
+		for ftype in MANIFEST2_IDENTIFIERS:
+						self.fhashdict[ftype] = {}
+		for profiledir, profiledir_dir, files in os.walk(profiledir):
+			for f in files:
+				try:
+					f = _unicode_decode(f,encoding=_encodings['fs'], errors='strict')
+					profiledir = _unicode_decode(profiledir,encoding=_encodings['fs'], errors='strict')
+				except UnicodeDecodeError:
+					continue
+				fpath = os.path.join(profiledir, f)
+				ftype = guessManifestFileType(fpath, is_pkg=False)
+				if ftype == "OTHER":
+					ftype = "DATA"
+				f = fpath.replace(self.pkgdir, "")
+				if not f.endswith("Manifest"):
+					self.fhashdict[ftype][f] = perform_multiple_checksums(fpath, self.hashes)
+
+	def checkFileHashes(self, ftype, fname, ignoreMissing=False, hash_filter=None):
+				digests = _filter_unaccelarated_hashes(self.fhashdict[ftype][fname])
+				if hash_filter is not None:
+						digests = _apply_hash_filter(digests, hash_filter)
+				try:
+						if ftype not in ('DIST'):
+								ok, reason = verify_all(self._getAbsname(ftype, fname), digests)
+								if not ok:
+										raise DigestException(tuple([self._getAbsname(ftype, fname)]+list(reason)))
+								return ok, reason
+				except FileNotFound as e:
+						if not ignoreMissing:
+								raise
+						return False, _("File Not Found: '%s'") % str(e)
+
+	def validate_signature(self):
+		'''Validate the Manifest's file signature.'''
+		self.gkeys = GkeysInterface('portage', '/')
+		is_good, has_sig = self.gkeys.verify_file(self.getFullname())
+		return (is_good, has_sig)
+
+	def verify_manifest(self, verify_pkgs = True):
+		'''Verifies the manifest's validity'''
+		mfdir = self.pkgdir
+		mftype = self.find_repotype(mfdir)
```

Comment:
I feel like we should be passing information into the Manifest constructor, at least the depth, in order to reduce the effort necessary for the Manifest object to orient itself withing the repository.